types/unist/index.d.ts

```diff
@@ -79,19 +78,21 @@ export interface Point {
     offset?: number | undefined;
 }
 
+export type NodeData<T extends Node> = T extends Node<infer R> ? R : never;
```

Comment:
What’s `R`?

Reply:
`Data` type extracted from `Node`, it's generic.

Reply:
Might be good to, where unclear, instead use multi-letter variable names

Reply:
`infer R` is very common to be used.

Reply:
:thinking: I've seen `R` used, but usually in the context of inferring a return type.
This isn't a return type, would it make sense to use:
```suggestion
export type NodeData<T extends Node> = T extends Node<infer Data> ? Data : never;
```
? :thinking:

Reply:
It's fine, that doesn't matter to me.